src/cart/cart-item.js

```diff
@@ -0,0 +1,33 @@
+import { Product } from "../product/product";
+
+export class CartItem {
+  #product;
+  #quantity;
+
+  constructor(product, quantity) {
+    if (!(product instanceof Product)) {
+      throw new Error("Product must be an instance of Product class.");
+    }
+    this.#product = product;
+    this.quantity = quantity;

```

Comment:
بین کدهایی که ربطی به هم ندارن، یه خط خالی بذارید که خوانایی بره بالاتر
مخصوصاً کدی که آکلاد داره، باید از بقیه جدا باشه چون یه Scope جداست

```suggestion
    if (!(product instanceof Product)) {
      throw new Error("Product must be an instance of Product class.");
    }
    
    this.#product = product;
    this.quantity = quantity;

```